frontend/src/app/api/user/update-tokens.ts

```diff
@@ -0,0 +1,56 @@
+import { createClient } from "@supabase/supabase-js";
+import { type NextRequest } from "next/server";
+
+export async function POST(request: NextRequest) {
+  const webhookSecret = request.headers.get('x-webhook-secret');
+  if (webhookSecret !== process.env.WEBHOOK_SECRET) {
+    return new Response(JSON.stringify({ error: 'Unauthorized' }), {
+      status: 403,
+      headers: { 'Content-Type': 'application/json' }
+    });
+  }
```

Comment:
Make sure to use NextResponse instead [[ref](https://nextjs.org/docs/app/api-reference/functions/next-response#json)]
```suggestion
  if (webhookSecret !== process.env.WEBHOOK_SECRET) {
    return NextResponse.json({ error: 'Unauthorized' }, { status: 403 });
  }
```